discovery/discovery_test.go

```diff
@@ -234,13 +252,11 @@ func TestNewDBOrchestratorPoolCache_GivenListOfOrchs_CreatesPoolCacheCorrectly(t
 	cachedOrchs, err := cacheDBOrchs(node, orchestrators)
 	require.Nil(err)
 	assert.Len(cachedOrchs, 3)
-	assert.Equal(cachedOrchs[1].ServiceURI, addresses[1])
```

Comment:
Why are these tests going away?

Reply:
If an `OrchInfo `request to` addresses[2]` returns before the one to` addresses[1]` then `addresses[2]` will be inserted into the database before `addresses[1]`

So the tests don't pass. Beforehand we were just looping over an array sequentially to correct them to the right format. Now we have an http request on a goroutine in between.

Reply:
Ah I see. Should we check that the orchs stored in the DB are actually the ones we expect then?

Reply:
Instead of checking 1 element we are now checking all elements are within the returned slices.  

```
	cachedOrchs, err := cacheDBOrchs(node, orchestrators)
	require.Nil(err)
	assert.Len(cachedOrchs, 3)
	for _, o := range orchestrators {
		dbO := ethOrchToDBOrch(o)
		dbO.PricePerPixel, _ = common.PriceToFixed(big.NewRat(999, 1))

		assert.Contains(cachedOrchs, dbO)
	}

	// ensuring orchs exist in DB
	orchs, err := node.Database.SelectOrchs(nil)
	require.Nil(err)
	assert.Len(orchs, 3)
	for _, o := range orchestrators {
		dbO := ethOrchToDBOrch(o)

		assert.Contains(orchs, dbO)
	}
```